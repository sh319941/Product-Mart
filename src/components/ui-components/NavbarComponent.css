   .hidden.menu {
      display: none;
    }

    .masthead.segment {
      min-height: 700px;
      padding: 1em 0;
    }
    .masthead .logo.item img {
      margin-right: 1em;
    }
    .masthead .ui.menu .ui.button {
      margin-left: 0.5em;
    }
 

    .ui.vertical.stripe {
      padding: 8em 0;
    }
    .ui.vertical.stripe .floated.image {
      clear: both;
    }
    .ui.vertical.stripe p {
      font-size: 1.33em;
    }
    .ui.vertical.stripe .horizontal.divider {
      margin: 3em 0;
    }

    .quote.stripe.segment {
      padding: 0;
    }
    .quote.stripe.segment .grid .column {
      padding-top: 5em;
      padding-bottom: 5em;
    }

    .footer.segment {
      padding: 5em 0;
    }

    .secondary.pointing.menu .toc.item {
      display: none;
    }

    @media only screen and (max-width: 700px) {
      .ui.fixed.menu {
        display: none !important;
      }
      .secondary.pointing.menu .item,
      .secondary.pointing.menu .menu {
        display: none;
      }
      .secondary.pointing.menu .toc.item {
        display: block;
      }
      .masthead.segment {
        min-height: 350px;
      }
    }
    .link, ddm {
    color:steelblue;
    font-weight: bold;
    font-size: 1.2rem;
  }
  .link:hover {
    text-decoration: none;
    font-weight: bolder;
    background: rgba(123, 123, 123, 0.5);
  }

  .link > a:hover {
    text-decoration: none;
  }
  
  .ddm {
    color: steelblue;
    font-weight: bold;
    font-size: 1.2rem;
  }

  .brand-link {
    font-weight: bolder;
    font-size: 1.5rem;
  }

  .brand-name {
    color: Black;
    font-weight: bolder;
    font-size: 1.5rem;
  }
  .brand-name:hover {
    text-decoration: none;
  }