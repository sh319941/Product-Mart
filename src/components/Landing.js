import React from 'react';
import { Link } from 'react-router-dom';

import NavbarComponent from './ui-components/NavbarComponent';
import './landing.css';

const Landing = () => {

    return (
      <div>
        <NavbarComponent />
        <div class="carouselDiv">
          <div
            id="carouselExampleCaptions"
            class="carousel slide"
            data-ride="carousel"
          >
            <ol class="carousel-indicators">
              <li
                data-target="#carouselExampleCaptions"
                data-slide-to="0"
                class="active"
              ></li>
              <li data-target="#carouselExampleCaptions" data-slide-to="1"></li>
              <li data-target="#carouselExampleCaptions" data-slide-to="2"></li>
            </ol>
            <div class="carousel-inner">
              <div class="carousel-item active">
                <img
                  src="https://www.reliablesoft.net/wp-content/uploads/2017/07/homepage-seo-ecommerce.jpg" height="550px" width ="100px"
                  class="d-block w-100"
                  alt="displayimg"
                />
                <div class="carousel-caption d-none d-md-block">
                  <h5>Online Shopping</h5>
                  <p>React Capstone Project</p>
                </div>
              </div>
              <div class="carousel-item">
                <img
                  src="https://www.pretty-pages.com/wp-content/uploads/2018/10/facebook-online-shopping-post.jpg" height="550px" width ="100px"
                  class="d-block w-100"
                  alt="displayimg"
                />
                <div class="carousel-caption d-none d-md-block">
                <h5>Online Shopping</h5>
                  <p>React Capstone Project</p>
                </div>
              </div>
              <div class="carousel-item">
                <img
                  src="https://mk0ethicalnetgpxgjkd.kinstacdn.com/wp-content/uploads/2019/11/thumbnail-onine-shopping.jpg" height="550px" width ="100px"
                  class="d-block w-100"
                  alt="displayimg"
                />
                <div class="carousel-caption d-none d-md-block">
                <h5>Online Shopping</h5>
                  <p>React Capstone Project</p>
                </div>
              </div>
            </div>
            <a
              class="carousel-control-prev"
              href="#carouselExampleCaptions"
              role="button"
              data-slide="prev"
            >
              <span
                class="carousel-control-prev-icon"
                aria-hidden="true"
              ></span>
              <span class="sr-only">Previous</span>
            </a>
            <a
              class="carousel-control-next"
              href="#carouselExampleCaptions"
              role="button"
              data-slide="next"
            >
              <span
                class="carousel-control-next-icon"
                aria-hidden="true"
              ></span>
              <span class="sr-only">Next</span>
            </a>
          </div>
        </div>
        <section class="text-center about">
          <h1>About US</h1>
          <div class="container">
            <div class="row">
              <div
                class="col-lg-4 col-sm-12 about-item wow lightSpeedIn"
                data-wow-offset="200"
              >
                <span class="fa fa-group"></span>
                <h2>Fully functional</h2>
                <p class="lead">
                  Login to app to view all features available in the platform
                </p>
              </div>
              <div
                class="col-lg-3 col-sm-12 about-item wow lightSpeedIn"
                data-wow-offset="200"
              >
                <span class="fa fa-info"></span>
                <h2>Statistics</h2>
                <p class="lead">
                  View the product statistics based on popularity
                </p>  
              </div>
              <div class="clearfix visible-md-block visible-sm-block"></div>
              <div
                class="col-lg-4 col-sm-6 col-ex-12 about-item wow lightSpeedIn"
                data-wow-offset="200"
              >
                <span class="fa fa-shopping-cart"></span>
                <h2>Shop products</h2>
                <p class="lead">
                  Select from a wide range of available products
                </p>
              </div>
            </div>
          </div>
        </section>
        <div className="container">
          <section>
           
          </section>
        </div>
        <footer className="bg-light">
          <div class="container">
            <div class="row">
              <div class="col-xs-6">
                <p class="copyright-text">
                  &copy; Product Mart 2020 All Rights Reserved
                  <br />
                  Designed by Shine Xavier.
                </p>
              </div>
           
            </div>
          </div>
        </footer>
      </div>
    );
}

export default Landing;