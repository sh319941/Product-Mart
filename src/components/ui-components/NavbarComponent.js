import React from 'react';
import { useHistory, Link } from "react-router-dom";
import { Navbar, Nav, NavDropdown } from "react-bootstrap";

import './NavbarComponent.css';

const NavbarComponent = () => {

  let history = useHistory();
  const renderAuthButton = () => {
    if (localStorage.getItem("username") === null) {
      return (
        <Navbar bg="light" expand="lg">
          <Navbar.Brand>
            <Link to="/" className="brand-name">
              Product Mart
            </Link>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="ml-auto">
              <Nav.Link className="link">
                <Link to="/products">Products</Link>
              </Nav.Link>

              <Nav.Link className="link">
                <Link to="/addProduct">Add Product</Link>
              </Nav.Link>

              <Nav.Link className="link">
                <Link to="/login">Login</Link>
              </Nav.Link>
              <Nav.Link className="link">
                <Link to="/register">Register</Link>
              </Nav.Link>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
      );
    } else {
      return (
        <Navbar bg="light" expand="lg">
          <Navbar.Brand>
            <Link to="/" className="brand-name">
              Product Mart
            </Link>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="ml-auto">
              <Nav.Link className="link">
                <Link to="/products">Products</Link>
              </Nav.Link>  
              <Nav.Link className="link">
                <Link to="/myproducts">My Products</Link>
              </Nav.Link>
              <Nav.Link className="link">
                <Link to="/addProduct">Add Product</Link>
              </Nav.Link>
              <Nav.Link className="link">
                <Link to="/cart">Cart</Link>
              </Nav.Link>
              <Nav.Link className="link">
                <Link to="/productStats">Statistics</Link>
              </Nav.Link>
              <NavDropdown
                title={localStorage.getItem("username")}
                id="basic-nav-dropdown"
                className="ddm"
              >
                <NavDropdown.Item>
                  <Link to="/myprofile">My Profile</Link>
                </NavDropdown.Item>
                <NavDropdown.Divider />
                <NavDropdown.Item onClick={() => logout()}>
                  Logout
                </NavDropdown.Item>
              </NavDropdown>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
      );
    }
  };

    function logout() {
      localStorage.clear();
      history.push("/");
    }

  return <div>{renderAuthButton()}</div>;
};

export default NavbarComponent;