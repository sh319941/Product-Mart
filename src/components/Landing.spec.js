import React from 'react';
import { shallow } from 'enzyme';

import Landing from './Landing';
import NavbarComponent from './ui-components/NavbarComponent';

describe("Landing Page", () => {

    it("Snapshot Of Landing Page", () => {
        const component = shallow(<Landing />);
        expect(component).toMatchSnapshot();
    });

  it("Check if  Navbar is rendered", () => {
    const wrapper = shallow(<Landing />);
    expect(wrapper.containsMatchingElement(<NavbarComponent />)).toEqual(true);
  });
  
});