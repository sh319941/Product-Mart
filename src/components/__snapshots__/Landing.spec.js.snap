// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page Snapshot Of Landing Page 1`] = `
<div>
  <NavbarComponent />
  <div
    class="carouselDiv"
  >
    <div
      class="carousel slide"
      data-ride="carousel"
      id="carouselExampleCaptions"
    >
      <ol
        class="carousel-indicators"
      >
        <li
          class="active"
          data-slide-to="0"
          data-target="#carouselExampleCaptions"
        />
        <li
          data-slide-to="1"
          data-target="#carouselExampleCaptions"
        />
        <li
          data-slide-to="2"
          data-target="#carouselExampleCaptions"
        />
      </ol>
      <div
        class="carousel-inner"
      >
        <div
          class="carousel-item active"
        >
          <img
            alt="displayimg"
            class="d-block w-100"
            height="550px"
            src="https://www.reliablesoft.net/wp-content/uploads/2017/07/homepage-seo-ecommerce.jpg"
            width="100px"
          />
          <div
            class="carousel-caption d-none d-md-block"
          >
            <h5>
              Online Shopping
            </h5>
            <p>
              React Capstone Project
            </p>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <img
            alt="displayimg"
            class="d-block w-100"
            height="550px"
            src="https://www.pretty-pages.com/wp-content/uploads/2018/10/facebook-online-shopping-post.jpg"
            width="100px"
          />
          <div
            class="carousel-caption d-none d-md-block"
          >
            <h5>
              Online Shopping
            </h5>
            <p>
              React Capstone Project
            </p>
          </div>
        </div>
        <div
          class="carousel-item"
        >
          <img
            alt="displayimg"
            class="d-block w-100"
            height="550px"
            src="https://mk0ethicalnetgpxgjkd.kinstacdn.com/wp-content/uploads/2019/11/thumbnail-onine-shopping.jpg"
            width="100px"
          />
          <div
            class="carousel-caption d-none d-md-block"
          >
            <h5>
              Online Shopping
            </h5>
            <p>
              React Capstone Project
            </p>
          </div>
        </div>
      </div>
      <a
        class="carousel-control-prev"
        data-slide="prev"
        href="#carouselExampleCaptions"
        role="button"
      >
        <span
          aria-hidden="true"
          class="carousel-control-prev-icon"
        />
        <span
          class="sr-only"
        >
          Previous
        </span>
      </a>
      <a
        class="carousel-control-next"
        data-slide="next"
        href="#carouselExampleCaptions"
        role="button"
      >
        <span
          aria-hidden="true"
          class="carousel-control-next-icon"
        />
        <span
          class="sr-only"
        >
          Next
        </span>
      </a>
    </div>
  </div>
  <section
    class="text-center about"
  >
    <h1>
      About US
    </h1>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-4 col-sm-12 about-item wow lightSpeedIn"
          data-wow-offset="200"
        >
          <span
            class="fa fa-group"
          />
          <h2>
            Fully functional
          </h2>
          <p
            class="lead"
          >
            Login to app to view all features available in the platform
          </p>
        </div>
        <div
          class="col-lg-3 col-sm-12 about-item wow lightSpeedIn"
          data-wow-offset="200"
        >
          <span
            class="fa fa-info"
          />
          <h2>
            Statistics
          </h2>
          <p
            class="lead"
          >
            View the product statistics based on popularity
          </p>
        </div>
        <div
          class="clearfix visible-md-block visible-sm-block"
        />
        <div
          class="col-lg-4 col-sm-6 col-ex-12 about-item wow lightSpeedIn"
          data-wow-offset="200"
        >
          <span
            class="fa fa-shopping-cart"
          />
          <h2>
            Shop products
          </h2>
          <p
            class="lead"
          >
            Select from a wide range of available products
          </p>
        </div>
      </div>
    </div>
  </section>
  <div
    className="container"
  >
    <section />
  </div>
  <footer
    className="bg-light"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-xs-6"
        >
          <p
            class="copyright-text"
          >
            Â© Product Mart 2020 All Rights Reserved
            <br />
            Designed by Shine Xavier.
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
